src/main/java/picard/sam/SamErrorMetric/CollectSamErrorMetrics.java

```diff
@@ -328,30 +416,205 @@ private void writeMetricsFileForAggregator(final BaseErrorAggregation locusAggre
     }
 
     /**
-     * Advance the iterator until at or ahead of locus. if there's a variant at the locus return true, otherwise false.
-     * <p>
-     * HAS SIDE EFFECTS!!! draws from vcfIterator!!!
+     * If there's a variant at the locus return true, otherwise false.
+     *
+     * HAS SIDE EFFECTS!!! Queries the vcfFileReader
      *
-     * @param vcfIterator        a {@link VariantContext} iterator to advance (assumed sorted)
+     * @param vcfFileReader      a {@link VCFFileReader} to query for the given locus
      * @param locus              a {@link Locus} at which to examine the variants
      * @param sequenceDictionary a dictionary with which to compare the Locatable to the Locus...
      * @return true if there's a variant over the locus, false otherwise.
      */
-    private static boolean advanceIteratorAndCheckLocus(final PeekableIterator<VariantContext> vcfIterator, final Locus locus, final SAMSequenceDictionary sequenceDictionary) {
-        while (vcfIterator.hasNext() && (vcfIterator.peek().isFiltered() ||
-                CompareVariantContextToLocus(sequenceDictionary, vcfIterator.peek(), locus) < 0)) {
-            vcfIterator.next();
+    private static boolean checkLocus(final VCFFileReader vcfFileReader, final Locus locus, final SAMSequenceDictionary sequenceDictionary) {
+
+        boolean overlaps = false;
+        final Interval queryInterval = locusToInterval(locus, sequenceDictionary);
+
+        if ( queryInterval != null ) {
+
+            try ( final CloseableIterator<VariantContext> vcfIterator = vcfFileReader.query(queryInterval) ) {
+
+                overlaps = true;
+
+                boolean allFiltered = true;
+
+                while ( vcfIterator.hasNext() ) {
+                    final VariantContext vcf = vcfIterator.next();
+                    if ( vcf.isFiltered() ) {
+                        continue;
+                    }
+                    allFiltered = false;
+                }
+
+                if ( allFiltered ) {
+                    overlaps = false;
+                }
+            }
+        }
+
+        return overlaps;
+    }
+
+    /**
+     * Converts the given locus into an interval using the given sequenceDictionary.
+     * @param locus The {@link Locus} to convert.
+     * @param sequenceDictionary The {@link SAMSequenceDictionary} to use to convert the given {@code locus}.
+     * @return An {@link Interval} representing the given {@code locus} or {@code null} if it cannot be converted.
+     */
+    private static Interval locusToInterval(final Locus locus, final SAMSequenceDictionary sequenceDictionary) {
```

Comment:
is there really no utility method for this? if so, please find an appropriate utils class and put it there.

Reply:
@jonn-smith I couldn't find a utility function for this, do you know of any or a good place to put it? (he wrote this)

Reply:
I wrote it because I couldn't find something that did this.  The closest place to put it seems to be in `IntervalListTools`.

Reply:
if you use Locatable instead of Locus, you'll be all set.